ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java

```diff
@@ -1306,6 +1308,122 @@ public void shouldFailTerminateOnInvalidDeleteTopicPattern() {
     ksqlResource.terminateCluster(request);
   }
 
+  @Test
+  public void shouldFailIfRegisterTopicAlreadyExists() {
+    // Given:
+    final String registerSql = "REGISTER TOPIC FOO WITH (kafka_topic='bar', value_format='json');";
+    metaStore.putTopic(new KsqlTopic("FOO", "foo", new KsqlJsonTopicSerDe(), true));
```

Comment:
You can just use a mock here for the topic.  You could even move this into a `givenTopicRegistered("foo")` method to aid readability of the test.  (I'd rename the existing `givenTopicExists` to a more explicit `givenTopicExistsInKafka` or similar.)

Reply:
`KsqlTopic` is essentially a POJO. I feel like it's overkill to mock something with no functionality. I will put it in a method called `givenTopicRegesitered` though